### ISSUE TITLE
Make the `sync` feature enabled by default.

### ISSUE THREAD

**Issue by ChillFish8** [meilisearch/meilisearch-rust#250]
Currently, you can enable the `sync` feature manually however, this doesn't massively make sense to have as an opt-in thing when it's originally targeting the Tokio ecosystem where most systems will likely be using the multi-threaded runtime.

In the case that they *are* using a single-threaded runtime and are at the point of wanting to save on the atomic cost, then they can opt-out. Although the overhead of the atomic operation is completely insignificant in reality and I'm not sure if it's worth having it as a feature, to begin with considering it is so cheap.

**Comment by brunoocasali** [comment]
Having a small conversation with @irevoire we agreed that is a good addition to the package!

Feel free to submit a PR @ChillFish8 😄

**Comment by irevoire** [comment]
Yeah, you can remove the feature entirely and use `Arc` by default everywhere :+1:
